core/coreapi/pin_test.go

```diff
@@ -0,0 +1,209 @@
+package coreapi_test
+
+import (
+	"context"
+	"strings"
+	"testing"
+)
+
+func TestPinAdd(t *testing.T) {
+	ctx := context.Background()
+	_, api, err := makeAPI(ctx)
+	if err != nil {
+		t.Error(err)
+	}
+
+	p, err := api.Unixfs().Add(ctx, strings.NewReader("foo"))
+	if err != nil {
+		t.Error(err)
+	}
+
+	err = api.Pin().Add(ctx, p)
```

Comment:
Should be verify the content was pined? e.g. `list, err = api.Pin().Ls(ctx) && len(list) == 1` as is done elsewhere?

Reply:
This is more of an example rather than a test, since this is tested a few times below I skipped that check here.